Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -791,55 +791,73 @@ public boolean chooseTargetAmount(Outcome outcome, TargetAmount target, Ability
             abilityControllerId = target.getAbilityController();
         }
 
-        while (canRespond()) {
-            Set<UUID> possibleTargets = target.possibleTargets(source == null ? null : source.getSourceId(), abilityControllerId, game);
-            boolean required = target.isRequired(source != null ? source.getSourceId() : null, game);
+        int amount = target.getAmountRemaining();
+        if (amount < 1) {
+            return false;
+        }
+
+        Map<UUID, String> targets = new LinkedHashMap<>(amount);
```

Comment:
Some target types can checks already selected targets and can fail on next `canTarget` call. So you must adds selected targets to real target object all the time, not in the end.